### ISSUE TITLE
confidence of binary decision is absent 

### ISSUE THREAD

**Issue by Sandy4321** [fchollet/keras#1992]
confidence for decision is very good to have

for example neural network gives decision 1 for given input data sample
let say that data sample is text, then text has positive sentiment
then if confidence is available it maybe used for estimation of reliability of this decision
 example 1
 decision is 1 , and confidence measured by probability is  0.99
so decision is reliable
 example 1
 decision is 1 , and confidence measured by probability is  0.1
so decision is un reliable

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [ x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [x ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by dbonadiman** [comment]
The output of the network isn't already a confidence score for classification?

I'm not following.

**Comment by Sandy4321** [comment]
output of NN is probability of correct decision
for example 
actual labels:  1  1  0  0  1  0  1  0  1  1
detected:        0  1  0  0  0  1  1  0  1  1
so number of correct decisions is 7
number of mistakes is 3
then probability of correct decision is  3/10 = 0.3, so code gives me this number 0.3, in lstm case is it is named accuracy

but if I know that for first test sample 1 was detects 0 with reliability  0.3 ,
then I may mark this sample as detected with low reliability
Related to this question is question how to get these decisions from NN
so I need these   
detected data:        0  1  0  0  0  1  1  0  1  1

**Comment by orsiluk** [comment]
I have the same problem. I would like to know the confidence level of the predicted class for each element from my array, not only the overall accuracy.
